### PR TITLE
Reinstate Python2 doctests with patched print function.

### PR DESCRIPTION
**Pull request by pp-mo**
A small improvement, if you like it @lbdreyer ?

### CODE REVIEW

File: docs/iris/src/userguide/interpolation_and_regridding.rst [hunk 1]
```diff
@@ -5,8 +5,23 @@
...
   import numpy as np
   import iris
+  # Mask out all warnings
   import warnings
   warnings.simplefilter('ignore')
+  # Patch print function for same numpy formatting in Python 2 and 3
+  from iris.util import _printopts_context
+  _old_print_func = print
+  def _new_print_func(x):
+      with _printopts_context(legacy='1.13'):
```
Comment (lbdreyer) [inline docs/iris/src/userguide/interpolation_and_regridding.rst]: I'm not so keen on the pinning to a version but at least this is consistent with the cml and ensures it's more stable :+1: 

I'll merge assuming the tests pass.

Comment (pp-mo) [reply docs/iris/src/userguide/interpolation_and_regridding.rst]: If you pass anything else in, it warns you that only '1.13' and None are currently accepted :frowning_face:

Comment (lbdreyer) [reply docs/iris/src/userguide/interpolation_and_regridding.rst]: I see, so it's not like there are multiple versions and we're pinning to one, rather we're enabling the legacy version of printing things which numpy has set to 1.13

**Comment by pp-mo** [comment]
Tricky, it seems the print function patching does not work in Python2, but it does if I try it in an interpreter.  Maybe something subtly about the way Sphinx works ??
I'm giving this up for now.

**Pull request closed without merging**